website/blog/2022-07-26-pre-commit-dbt.md

```diff
@@ -0,0 +1,274 @@
+
+In our example, let’s just start by saying that we want:
+- all our models to have been added to a YAML file
+- all our models to include some tests
+
+To configure pre-commit, we have to create a file called “.pre-commit-config.yaml” at the root of our dbt project and add the following content:
+
+```yaml
+repos:
+- repo: https://github.com/offbi/pre-commit-dbt
+  rev: v1.0.0
+  hooks:
+  - id: dbt-docs-generate
+  - id: check-model-has-properties-file
+    name: Check that all models are listed in a YAML file
+    files: ^models/
+  - id: check-model-has-tests
+    name: Check that all models have tests
+    files: ^models/
+```
+
+A few notes about this file:
+- We start with a dbt-docs-generate to ensure that the command “dbt docs generate” is run before checking our models. This is required because pre-commit-dbt reads from the artifacts `catalog.json` and `manifest.json` and those files can be generated by generating the documentation
+- The `files` argument is a regular expression. `^models/` is going to apply the test to all our models whose path starts with “models”, i.e. all the models of our project. If we wanted to run the test only for the models in the mart folder, we could write `^models/mart` . There are a few handy tools online to define and test regular expressions; one of them is https://regex101.com/
+- We could also provide values for the parameter `exclude` if we wanted to exclude specific files or folders
+- We could have added a parameter to the hook `check-model-has-tests` to mention how many tests each model is supposed to have (see [here](https://github.com/offbi/pre-commit-dbt/blob/main/HOOKS.md#check-model-has-tests))
+
+Now that we have defined our configuration file, the next step will depend on whether we are using dbt via dbt Cloud or dbt Core via the CLI.
+
+If we are using dbt Cloud, let’s jump to step 4, where we will set up pre-commit-dbt as part of the CI process, otherwise, with dbt Core we can go to step 2.
+
+### 2. Understand the scope of the changes required
+
+With our configuration file created, we can now set up our computer to do some checks locally.
+
+#### Activating a Python virtual environment
+
+If we are using dbt in a Python virtual environment, let’s activate this environment. If not, we should really create a Python virtual environment ([more info here](https://docs.python.org/3/library/venv.html)) and activate it before installing pre-commit.
+
+We could technically skip this step but might then end up getting issues on our computer with different Python packages conflicting with each other.
+
+#### Installing and running pre-commit
+
+Once in the Python virtual environment, installing `pre-commit` is as straightforward as running `pip install pre-commit`.
```

Comment:
I am personally not doing it 😁 but I am happy to update the instructions to replace `pip` by `python -m pip`

Reply:
In case you haven't seen any explanations _why_ `python -m pip` is the recommended command:
- Longer:[ Why you should use `python -m pip`](https://snarky.ca/why-you-should-use-python-m-pip/)
- Shorter:[ Use 'python -m pip' Everywhere](https://adamj.eu/tech/2020/02/25/use-python-m-pip-everywhere/)